### ISSUE TITLE
Add categories

### ISSUE THREAD

**Issue by jtpio** [voila-gallery/voila-gallery.github.io#16]
The gallery could group the examples by category.

For example there could be a category to showcase widget libraries, and another one to showcase more complex examples.

**Comment by jtpio** [comment]
cc @martinRenou if you have some input on that

**Comment by martinRenou** [comment]
I agree it would be nice. Should we discuss the reordering of the examples in this issue as well or should we open another one?

**Comment by jtpio** [comment]
I guess here makes sense? We would do the change at the same time (adding the categories and reordering the examples)

**Comment by jtpio** [comment]
Also we could find a way to integrate this one nicely: https://github.com/voila-gallery/voila-gallery.github.io/issues/6

**Comment by martinRenou** [comment]
Maybe the category thing is a bit early? I mean we don't have that many examples yet. I guess we need a bit more.

We could have an:
- interactive widgets section like you said
- custom templates section, showcasing the vuetify default template, the material one, maybe the adminlte one. And showing the themes support.
- other libraries like bokeh/plotly...

**Comment by jtpio** [comment]
> Maybe the category thing is a bit early? I mean we don't have that many examples yet. I guess we need a bit more.

Good point, they almost still all fit on a high resolution screen at the moment.

**Comment by jtpio** [comment]
- And another category with advanced use cases (more complex apps)

**Comment by raphaelfournier** [comment]
Hello,
My screen probably has a lower resolution than yours, @jtpio, I felt that it would be nice to have a way to filter the displayed notebooks. I've created the #30 pull request, as a small proof of concept. I just adapted a w3school tuto. It's one of my first PR on GH, let me know if I may improve things for you before a merge (another branch, documentation, etc.).

I chose python-related names for place-holder categories, they should obvioulsy be adapted for a production version on the final voila gallery website.

**Comment by jtpio** [comment]
Thanks @raphaelfournier!

> I chose python-related names for place-holder categories, they should obvioulsy be adapted for a production version on the final voila gallery website.

This repo corresponds the voila gallery website: https://voila-gallery.org/

When a PR is merged it is automatically deployed on GitHub pages a few minutes after. So we could choose real names for the categories.
